src/core-sdk-tasks/ExtractArchiveToDirectory.cs

```diff
@@ -73,13 +78,41 @@ public override bool Execute()
             {
                 if (ValidateParameters())
                 {
+                    if (DirectoriesToCopy != null && DirectoriesToCopy.Length != 0)
+                    {
+                        var zip = new ZipArchive(File.OpenRead(SourceArchive));
+                        string loc = DestinationDirectory;
+                        foreach (var entry in zip.Entries)
+                        {
+                            foreach (var directory in DirectoriesToCopy.Split(';'))
+                            {
+                                if (entry.FullName.Contains(directory))
+                                {
+                                    string dirBuilder = loc;
+                                    foreach (var pathPart in Path.GetDirectoryName(entry.FullName).Split('/'))
+                                    {
+                                        if (!Directory.Exists(Path.Combine(dirBuilder, pathPart)))
+                                        {
+                                            Directory.CreateDirectory(Path.Combine(dirBuilder, pathPart));
```

Comment:
CreateDirectory creates all the subdirs, so I don't think this is necessary?

Reply:
Great, I didn't know this. I will simplify this code.